backend/src/apportionment/mod.rs

```diff
@@ -408,22 +532,25 @@ fn step_allocate_remainder_using_highest_surplus(
             political_groups_qualifying_for_unique_highest_average(assigned_seats, previous)
                 .peekable();
         if qualifying_for_unique_highest_average.peek().is_some() {
+            // TODO: Can I somehow call step_allocate_remainder_using_highest_averages with
+            //  qualifying_for_unique_highest_average to prevent duplicate code?
             let selected_pgs = political_groups_with_largest_average(
                 qualifying_for_unique_highest_average,
-                remaining_seats,
+                residual_seats,
             )?;
```

Comment:
I'm not completely sure what you mean with this TODO, something like this?
```suggestion
        if let Some(assigned_seats) = qualifying_for_unique_highest_average.peek() {
            let selected_pgs = political_groups_with_largest_average(
                assigned_seats,
                residual_seats,
            )?;
```